frontend/src/CSets.js

```diff
@@ -223,114 +298,82 @@ function ConceptSetsPage(props) {
-        'filter=codesetId:' + codesetIds.join('|')
-      ].join('&')
+  let codesetIds = (qsParams && qsParams.codesetId && qsParams.codesetId.sort()) || []
+  let url = backend_url('concept-sets-with-concepts?concept_field_filter=conceptId&concept_field_filter=conceptName&codeset_id=' + codesetIds.join('|'))
 
   const { isLoading, error, data, isFetching } = useQuery([url], () => {
-    //if (codesetIds.length) {
-    //   console.log('fetching backend_url', url)
-      return axios.get(url).then((res) => res.data)
-      // console.log('enclave_url', enclave_url('objects/OMOPConceptSet'))
-      // .then((res) => res.data.data.map(d => d.properties))
-    //} else {
-      //return {isLoading: false, error: null, data: [], isFetching: false}
-    //}
-  });
-  // console.log('rowData', data)
-  return  <div>
-            <h4>Concepts:</h4>
-            <Table rowData={data} />
-          </div>
-  /*
-  let params = useParams();
-  let {conceptId} = params;
-  let path = `objects/OMOPConceptSet/${conceptId}/links/omopconcepts`;
-  let url = enclave_url(path)
-  const { isLoading, error, data, isFetching } = useQuery([path], () =>
-      axios
-          .get(url)
-          .then((res) => res.data.data.map(d => d.properties)) )
-  */
-}
+    return axios.get(url).then((res) => {
+      let concepts2darr = res.data.map((row) => {return Object.keys(row.concepts)});
+      let conceptsArr = [].concat(...concepts2darr);
+      let conceptsSet = [...new Set(conceptsArr)];
+      // todo: o(1) instead of o(n) would be better; like a python dict instead of array
+      let csetIdConcepts = res.data.map((row) => {return {codesetId: row.codesetId, conceptIds: Object.keys(row.concepts)}});
+      let tableData = conceptsSet.map((conceptId) => {return {'ConceptID': conceptId}});
 
+      // todo: o(1) instead of o(n) would be better; like a python dict instead of array
+      // Iterate over sets and put info in rows
+      let tableData2 = []
+      for (let row of tableData) {
+        let newRow = row;
+        for (let {codesetId, conceptIds} of csetIdConcepts) {
+          newRow[codesetId] = 'X';
+          for (let conceptId of conceptIds) {
+            if (conceptId === row.ConceptID) {
+              newRow[codesetId] = 'O';
+              break;
+            }
+          }
+        }
+        // TODO: Reorder cols here if possible (
+        //  -this doesn't work. even when i play around in node. even some responses say it doesn't work, even though
+        //  the response i took this example from has most votes
+        //  https://stackoverflow.com/questions/6959817/changing-the-order-of-the-object-keys
+        const objectOrder = {'ConceptID': null}
+        newRow = Object.assign(objectOrder, newRow);
+        tableData2.push(newRow);
+      }
+      
+      return tableData2
+  })});
 
-export {ConceptSetsPage, CsetSearch, ConceptList};
+  // TODO: How to pass column order? could I just put 'ConceptID' as the only column thta mtters
```

Comment:
@Sigfried I'd really like to figure this out:
> `  // TODO: How to pass column order? could I just put 'ConceptID' as the only column that matters`

I visited here https://www.ag-grid.com/react-data-grid/column-definitions/ , and I saw the examples, but when I tried to replicate, it didn't work. 'ConceptID' still keeps getting put last. but it needs to be first.

### edit
i think i figured it out. i'm passing the column def to the AGGrid wrapper component I made, not the actual AGGrid component

Reply:
Also, for the columns, I think leaving the IDs of csets as the column header is nice, but it might be nice to have some way to know the name. Maybe a mouseover, or a 'key' for the header meanings somewhere.